index.js

```diff
@@ -37,7 +37,7 @@ module.exports = function retryAsPromised(callback, options) {
       }, options.timeout);
     }
 
-    Promise.resolve(callback()).then(resolve).tap(function () {
+    Promise.resolve(callback({ $current: options.$current })).then(resolve).tap(function () {
```

Comment:
`$current` is because it's internal, if exposing it as an arg to the callback just call it `current`